ts/src/pro/bybit.ts

```diff
@@ -980,6 +984,150 @@ export default class bybit extends bybitRest {
+         * @description watch all open positions
+         * @param {[string]|undefined} symbols list of unified market symbols
+         * @param {object} params extra parameters specific to the bybit api endpoint
+         * @returns {[object]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
+         */
+        await this.loadMarkets ();
+        const method = 'watchPositions';
+        let messageHash = '';
+        if (!this.isEmpty (symbols)) {
+            symbols = this.marketSymbols (symbols);
+            messageHash = '::' + symbols.join (',');
+        }
+        const firstSymbol = this.safeString (symbols, 0);
+        const url = this.getUrlByMarketType (firstSymbol, true, method, params);
+        messageHash = 'positions' + messageHash;
+        const client = this.client (url);
+        await this.authenticate (url);
+        this.setPositionsCache (client, symbols);
+        const cache = this.positions;
+        const fetchPositionsSnapshot = this.handleOption ('watchPositions', 'fetchPositionsSnapshot', true);
+        const awaitPositionsSnapshot = this.safeValue ('watchPositions', 'awaitPositionsSnapshot', true);
+        if (fetchPositionsSnapshot && awaitPositionsSnapshot && cache === undefined) {
+            const snapshot = await client.future ('fetchPositionsSnapshot');
+            return this.filterBySymbolsSinceLimit (snapshot, symbols, since, limit, true);
+        }
+        const topics = [ 'position' ];
+        const newPositions = await this.watchTopics (url, messageHash, topics, 'position', params);
+        if (this.newUpdates) {
+            return newPositions;
+        }
+        return this.filterBySymbolsSinceLimit (cache, symbols, since, limit, true);
+    }
+
+    setPositionsCache (client: Client, symbols: string[] = undefined) {
+        if (this.positions !== undefined) {
+            return this.positions;
+        }
+        const fetchPositionsSnapshot = this.handleOption ('watchPositions', 'fetchPositionsSnapshot', true);
+        if (fetchPositionsSnapshot) {
+            const messageHash = 'fetchPositionsSnapshot';
+            if (!(messageHash in client.futures)) {
+                client.future (messageHash);
+                this.spawn (this.loadPositionsSnapshot, client, messageHash, symbols);
+            }
+        } else {
+            this.positions = new ArrayCacheBySymbolBySide ();
+        }
+    }
+
+    async loadPositionsSnapshot (client, messageHash, symbols) {
+        // as only one ws channel gives positions for all types, for snapshot must load all positions
+        const fetchFunctions = [
+            this.fetchPositions (symbols, { 'type': 'swap', 'subType': 'linear' }),
+            this.fetchPositions (symbols, { 'type': 'swap', 'subType': 'inverse' }),
+            this.fetchPositions (symbols, { 'type': 'option' }),
+        ];
```

Comment:
not sure if we should load them all by default, let's say you provide a single symbol only, why call `fetchPositions` three times? does not seem to make much sense

Reply:
@carlosmiei  Yes, this is slightly a weird behavior. The reason for this is there is one websocket channel for all positions, while 3 rest endpoints to get the snapshot for positions. So if we would only choose one fetchPosition, we would receive updates from the other kinds of positions but have an incomplete snapshot.

In the case the user requests one symbol I think we should still do this, as if they do a second watch call with a symbol of a different type, the subscription would be the same so it would assume the snapshot exists for that type but it would be incomplete.

The only alternative I can think of would be:
- Create one future for each type of snapshot
- If all symbols are watched, get all 3 snapshots
- If one symbol is passed, check if the snapshot has already been fulfilled if not get the missing snapshot
However with these solution I think it's a bit more complex and also we wouldn't be able to respect if the user had set `awaitPositionsSnapshot:true`